sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/condition/OnAnyPropertyCondition.java

```diff
@@ -76,10 +79,31 @@ private String[] getNames(Map<String, Object> annotationAttributes) {
             return (value.length > 0) ? value : name;
         }
 
+        private Set<String> getPrefixes(AnnotationAttributes annotationAttributes) {
+            Set<String> prefixSet = new HashSet<>();
+            prefixSet.add(annotationAttributes.getString("prefix"));
+            prefixSet.addAll(Arrays.asList(annotationAttributes.getStringArray("prefixes")));
+            return prefixSet.stream()
+                .filter(item -> !item.isEmpty())
+                .map(p -> AzureStringUtils.ensureEndsWithSuffix(p.trim(), PROPERTY_SUFFIX))
+                .collect(Collectors.toSet());
+        }
+
         private void collectProperties(PropertyResolver resolver, List<String> missing, List<String> nonMatching,
                                        List<String> matching) {
+            if (this.prefixes.size() > 0) {
+                for (String prefix : this.prefixes) {
+                    collectPropertiesWithPrefix(prefix, resolver, missing, nonMatching, matching);
+                }
+            } else {
+                collectPropertiesWithPrefix("", resolver, missing, nonMatching, matching);
+            }
```

Comment:
we can improve this, if the prefixes set is empty, we can add `""` to it, so the if/else won't be needed.